src/pasta/hashtocurve.rs

```diff
@@ -1,32 +1,64 @@
 //! This module implements "simplified SWU" hashing to short Weierstrass curves
 //! with a = 0.
 
+use std::convert::TryInto;
 use subtle::ConstantTimeEq;
 
 use crate::arithmetic::{Curve, CurveAffine, FieldExt};
 
 /// Hashes over a message and writes the output to all of `buf`.
 pub fn hash_to_field<F: FieldExt>(message: &[u8], domain_separation_tag: &[u8], buf: &mut [F]) {
```

Comment:
Does this function now no longer support filling the `buf` with random elements no matter its size? (My understanding is that is what the previous code did.) If now it only supports writing two elements to `buf`, then `buf` should change to `&mut [F; 2]`.